ksqldb-engine/src/main/java/io/confluent/ksql/engine/EngineExecutor.java

```diff
@@ -176,9 +176,10 @@ PullQueryResult executeTablePullQuery(
       throw new IllegalArgumentException("Executor can only handle pull queries");
     }
     final SessionConfig sessionConfig = statement.getSessionConfig();
-    PullSourceType sourceType = null;
-    PullPhysicalPlanType planType = null;
-    RoutingNodeType routingNodeType = null;
+    final RoutingNodeType routingNodeType = routingOptions.getIsSkipForwardRequest()
+        ? RoutingNodeType.REMOTE_NODE : RoutingNodeType.SOURCE_NODE;
```

Comment:
pulled this up because it cannot throw an exception and doesn't depend on the plan.